src/java/org/apache/cassandra/utils/bytecomparable/ByteComparable.java

```diff
@@ -188,30 +210,7 @@ static int length(ByteComparable src, Version version)
      */
     static int compare(ByteComparable bytes1, ByteComparable bytes2, Version version)
```

Comment:
I still do wonder why we really need to pass a particular `Version` here.
Maybe now that we have `Preencoded` we could infer the version.
If any of the arguments is instance of `Preencoded` get the version from there.
If both arguments are preencoded and the version doesn't match -> we can't compare them, throw.
If non of the arguments is preencoded, then we can just use the newest version; as I think any non-preencoded ByteComparable should work with the newest version.

WDYT?

Reply:
There's a new `compare` method for `Preencoded` that seems to align with what you are asking. This one was originally only meant for tests, I think we can now change SAI to fully use `Preencoded` and thus avoid the version here.

However, it seems we need this patch committed today, opened [CNDB-13504](https://github.com/riptano/cndb/issues/13504) for this work.